docs/javascripts/downloads.js

```diff
@@ -0,0 +1,191 @@
+    const services = [
+      {
+        service: '<a href="../services/qix-engine/introduction/">Qlik Associative Engine</a>',
+        feature: 'The powerful associative indexing engine from Qlik and the foundation of Qlik Core.',
+        link: '<a href="https://hub.docker.com/r/qlikcore/engine">qlikcore/engine</a>',
+        apis: [
+          'EngineAPI',
+          'EngineRestAPI',
+          'ScriptLanguageAPI',
+        ],
+      }, {
+        service: '<a href="../services/licenses/">Licenses</a>',
+        feature: 'Service required to run Qlik Associative Engine with a paid license',
+        link: '<a href="https://hub.docker.com/r/qlikcore/licenses">qlikcore/licenses</a>',
+        apis: [
+          'LicenseServiceAPI',
+        ],
+      }, {
+        service: '<a href="../services/mira/">Mira</a>',
+        feature: 'Qlik Associative Engine discovery service.',
+        link: '<a href="https://hub.docker.com/r/qlikcore/mira">qlikcore/mira</a><br>(source code: <a href="https://github.com/qlik-oss/mira">qlik-oss/mira</a>)',
+        apis: [
+          'MiraAPI',
+        ],
+      },
+    ];
+
+    const loader = document.createElement('div');
+    loader.className = 'dots';
+    for (let i = 0; i < 7; i++) {
+      const dot = document.createElement('div');
+      dot.className = 'dots__dot';
+      loader.appendChild(dot);
+    }
+
+    const scrollWrap = document.createElement('div');
+    scrollWrap.className = 'md-typeset__scrollwrap';
+
+    const tableContainer = document.createElement('div');
+    tableContainer.className = 'md-typeset__table';
+    scrollWrap.appendChild(tableContainer);
+
+    const table = document.createElement('table');
+    tableContainer.appendChild(table);
+
+    const tableHeader = document.createElement('thead');
+    table.appendChild(tableHeader);
+
+    const tableHeaderRow = document.createElement('tr');
+    tableHeader.appendChild(tableHeaderRow);
+    ['Service', 'Feature', 'Links', 'Latest Version'].forEach(title => {
+      const cell = document.createElement('th');
+      cell.appendChild(document.createTextNode(title));
+      tableHeaderRow.appendChild(cell);
+    });
+
+    const tableBody = document.createElement('tbody');
+    table.appendChild(tableBody);
+
+    const latestVersionCells = [];
+    services.forEach(service => {
+      const tableBodyRow = document.createElement('tr');
+      tableBody.appendChild(tableBodyRow);
+      ['service', 'feature', 'link'].forEach(column => {
+        const cell = document.createElement('td');
+        cell.insertAdjacentHTML('beforeend', service[column]);
+        tableBodyRow.appendChild(cell);
+      });
+
+      const cell = document.createElement('td');
+      cell.appendChild(loader.cloneNode(true));
+      tableBodyRow.appendChild(cell);
+      latestVersionCells.push(cell);
+    })
+    downloadsTableWrapper.appendChild(scrollWrap);
+
+    fetch('https://unpkg.com/enigma.js/schemas/12.34.11.json')
```

Comment:
Since we're using so few of the QIX API calls, we could define the schema ourselves and skip this HTTP request.

```
{
	"structs": {
        "Global": {
            "OpenDoc": {
				"In": [{ "Name": "qDocName","DefaultValue": "" }, { "Name": "qUserName","DefaultValue": "", "Optional": true }, { "Name": "qPassword","DefaultValue": "", "Optional": true }, { "Name": "qSerial","DefaultValue": "", "Optional": true }, { "Name": "qNoData","DefaultValue": false, "Optional": true }],
				"Out": []
			}
        },
        "Doc": {
            "CreateSessionObject": {
				"In": [{ "Name": "qProp","DefaultValue": {"qInfo": {"qId": "", "qType": ""}, "qExtendsId": "", "qMetaDef": {}} }],
				"Out": []
			},
        },
        "GenericObject": {
            "GetLayout": {
				"In": [],
				"Out": [{ "Name": "qLayout" }]
			}
        }
    }
}
```